### PR TITLE
Fix genome match

### PR DESCRIPTION
**Pull request by samirelanduk**
This is the proposed fix for the genome switching issue we had. There are a few checks/changes that might be needed before merging:

- Marc could you check the changes I made to primaryclipanalysis.nf? I basically took the pattern you used to get TRIMGALORE to work and applied it everywhere a channel had two inputs that needed to be kept in line with each other.
- I’ve added a `--method unique` to DEDUP processes - I gather from Charlotte that this needs to be both generated from the file contents as well as being overridable, so this will need to be modified.
- I also gather there are some CLIPPY changes to be made?

### CODE REVIEW

File: subworkflows/ncrna.nf [hunk 1]
```diff
@@ -8,57 +8,34 @@ include { STAR_ALIGN } from '../modules/nf-core/modules/star/align/main'
...
-    longest_transcript_index = genome.map{ folder -> file(folder + "/LONGEST_TRANSCRIPT/*.fa.fai")}
-    segmentation_gtf = genome.map{ folder -> file(folder + "/RAW_ICOUNT_SEGMENT/*segmentation*")}
-    regions_gtf = genome.map{ folder -> file(folder + "/RAW_ICOUNT_SEGMENT/*regions*")}
-
-    // Start things off with TrimGalore
-    TRIMGALORE ( reads )
-
-    // Run Bowtie Align on each of the reads, with the provided genome index
-    // files as reference
-    BOWTIE_ALIGN (
```
Comment (marc-jones) [inline subworkflows/ncrna.nf:50]: Deleted but not replaced in the workflow

File: subworkflows/ncrna.nf [hunk 2]
```diff
@@ -8,57 +8,34 @@ include { STAR_ALIGN } from '../modules/nf-core/modules/star/align/main'
...
 
-    // Run STAR Align on the reads which didn't match above
-    STAR_ALIGN (
-        BOWTIE_ALIGN.out.fastq,
-        star_index,
-        genome_gtf,
-        false,                   // star_ignore_sjdbgtf
-        "",                      // seq_platform
-        ""                       // seq_center
-    )
```
Comment (marc-jones) [inline subworkflows/ncrna.nf:63]: Deleted but not replaced in the workflow

File: subworkflows/ncrna.nf [hunk 3]
```diff
@@ -8,57 +8,34 @@ include { STAR_ALIGN } from '../modules/nf-core/modules/star/align/main'
...
-
-    // Start things off with TrimGalore
-    TRIMGALORE ( reads )
-
-    // Run Bowtie Align on each of the reads, with the provided genome index
-    // files as reference
-    BOWTIE_ALIGN (
-        TRIMGALORE.out.reads,
-        bowtie_index
-    )
```
Comment (marc-jones) [inline subworkflows/ncrna.nf:53]: Deleted but not replaced in the workflow

File: subworkflows/primaryclipanalysis.nf [hunk 4]
```diff
@@ -100,39 +151,88 @@ workflow PRIMARY_CLIP_ANALYSIS {
     STAR_SAMTOOLS_INDEX ( STAR_ALIGN.out.bam_sorted )
     ch_umi_input = STAR_ALIGN.out.bam_sorted.combine(STAR_SAMTOOLS_INDEX.out.bai, by: 0)
     UMITOOLS_DEDUP ( ch_umi_input )
-
     UMITOOLS_SAMTOOLS_INDEX ( UMITOOLS_DEDUP.out.bam )
+    reads.map{triplet -> [
+        triplet[0], file(triplet[2] + "/SAMTOOLS_FAIDX/*.fa.fai")
+    ]}.set{ ch_genome_fa }
```
Comment (marc-jones) [inline subworkflows/primaryclipanalysis.nf:157]: Tiny detail, but should be `ch_genome_fai`

File: subworkflows/primaryclipanalysis.nf [hunk 5]
```diff
@@ -100,39 +151,88 @@ workflow PRIMARY_CLIP_ANALYSIS {
...
 
-
     // CLIPPY Peak Calling
-    CLIPPY ( GET_CROSSLINKS.out.crosslinkBed, genome_gtf, genome_fai )
-
+    GET_CROSSLINKS.out.crosslinkBed.join( ch_gtf ).join( ch_genome_fa ).set{ ch_crosslinks }
+    ch_crosslinks.multiMap { tuple ->
+        crosslinks: [tuple[0], tuple[1]]
+        gtf: tuple[2]
+        fa: tuple[3]
```
Comment (marc-jones) [inline subworkflows/primaryclipanalysis.nf:173]: Likewise, should be `fai`

**Review by marc-jones — COMMENTED** [review]
Apart from the changes to `subworkflows/ncrna.nf` which changes the functionality, the rest looks good. I didn't anticipate the issues with having different references running simultaneously!

File: subworkflows/primaryclipanalysis.nf (hunk 4 above)
Comment (samirelanduk) [reply subworkflows/primaryclipanalysis.nf]: Fixed!

**Pull request merged**